From 3cbd98f4461e42dbfc1d9af0e6215fe6c78bb0d1 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 13 Aug 2017 18:37:06 +0300
Subject: [PATCH 1901/2241] net: mvpp2x: remove GoP padding configuration

This patch remove unneeded GoP padding configuration.
Padding is enabled by default and should be always on.

GoP padding were never changed since offset instead of mask
were used in register configuration and other register area
were overrun.
Issue didn't caused any seen damage.

Change-Id: Ie483c228e576b56d1d3ef0bd090c2ba467c332c1
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42942
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
index 062f339..a0b410d 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_gop110_hw.c
@@ -174,7 +174,6 @@ static void mv_gop110_gmac_rgmii_cfg(struct gop_hw *gop, int mac_num)
 
 	val = mv_gop110_gmac_read(gop, mac_num, MV_GMAC_PORT_CTRL2_REG);
 	val |= MV_GMAC_PORT_CTRL2_CLK_125_BYPS_EN_MASK;
-	val &= ~MV_GMAC_PORT_CTRL2_DIS_PADING_OFFS;
 	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL2_REG, val);
 
 	val = mv_gop110_gmac_read(gop, mac_num, MV_GMAC_PORT_CTRL0_REG);
@@ -214,10 +213,6 @@ static void mv_gop110_gmac_qsgmii_cfg(struct gop_hw *gop, int mac_num)
 	val &= ~MV_GMAC_PORT_CTRL4_QSGMII_BYPASS_ACTIVE_MASK;
 	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL4_REG, val);
 
-	val = mv_gop110_gmac_read(gop, mac_num, MV_GMAC_PORT_CTRL2_REG);
-	val &= ~MV_GMAC_PORT_CTRL2_DIS_PADING_OFFS;
-	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL2_REG, val);
-
 	val = mv_gop110_gmac_read(gop, mac_num, MV_GMAC_PORT_CTRL0_REG);
 	/* configure GIG MAC to SGMII mode */
 	val &= ~MV_GMAC_PORT_CTRL0_PORTTYPE_MASK;
@@ -256,7 +251,6 @@ static void mv_gop110_gmac_sgmii_cfg(struct gop_hw *gop, int mac_num)
 	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL4_REG, val);
 
 	val = mv_gop110_gmac_read(gop, mac_num, MV_GMAC_PORT_CTRL2_REG);
-	val |= MV_GMAC_PORT_CTRL2_DIS_PADING_OFFS;
 	val &= ~MV_GMAC_PORT_CTRL2_FC_MODE_MASK;
 	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL2_REG, val);
 
@@ -297,10 +291,6 @@ static void mv_gop110_gmac_sgmii2_5_cfg(struct gop_hw *gop, int mac_num)
 	val |= MV_GMAC_PORT_CTRL4_QSGMII_BYPASS_ACTIVE_MASK;
 	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL4_REG, val);
 
-	val = mv_gop110_gmac_read(gop, mac_num, MV_GMAC_PORT_CTRL2_REG);
-	val |= MV_GMAC_PORT_CTRL2_DIS_PADING_OFFS;
-	mv_gop110_gmac_write(gop, mac_num, MV_GMAC_PORT_CTRL2_REG, val);
-
 	val = mv_gop110_gmac_read(gop, mac_num, MV_GMAC_PORT_CTRL0_REG);
 	/* configure GIG MAC to 1000Base-X mode connected to a
 	 * fiber transceiver
-- 
2.7.4

